src/main.c

```diff
@@ -94,326 +98,405 @@ int exec_cmdline_arg(int argcin, char *argvin[])
 	unsigned long long size = 0;
 	bool json_flag = FALSE;
 	bool header_flag = TRUE;
-	unsigned char device[NAMELEN] = {0}, backing[NAMELEN] = {0}, host[NAMELEN] = {0}, header[NAMELEN] = {0};
+	char device[NAMELEN] = {0}, backing[NAMELEN] = {0}, host[NAMELEN] = {0}, header[NAMELEN] = {0};
+	char generic_msg[NAMELEN] = {0};
 	struct RD_PROFILE *disk = NULL;
 	struct RC_PROFILE *cache = NULL;
 	struct MEM_PROFILE *mem = NULL;
 	struct VOLUME_PROFILE *volumes = NULL;
 
 	sprintf(header, "%s %s\n%s\n\n", PROCESS, VERSION_NUM, COPYRIGHT);
 
-	while ((i = getopt(argcin, argvin, "a:b:c:d:ef:gH:hi:jL:lm:NnP:p:qRr:s:t:U:u:VvXx")) != INVALID_VALUE) {
+	while ((i = getopt(argcin, argvin, "?:a:b:c:d:ef:gH:hi:jL:lm:NnP:p:qRr:s:t:U:u:VvXx")) != INVALID_VALUE) {
 		switch (i) {
-		case 'h':
-			printf("%s", header);
-			online_menu(argvin[0]);
-			return SUCCESS;
-			break;
-		case 'a':
-			action = ACTION_ATTACH;
-			size = strtoul(optarg, (char **)NULL, 10);
-			break;
-		case 'b':
-			sprintf(backing, "%s", optarg);
-			break;
-		case 'c':
-			size = strtoul(optarg, (char **)NULL, 10);
-			break;
-		case 'd':
-			action = ACTION_DETACH;
-			sprintf(device, "%s", optarg);
-			break;
-		case 'e':
-			action = ACTION_EXPORT_NVMET;
-			break;
-		case 'f':
-			action = ACTION_FLUSH;
-			sprintf(device, "%s", optarg);
-			break;
-		case 'g':
-			header_flag = FALSE;
-			break;
-		case 'H':
-			sprintf(host, "%s", optarg);
-			break;
-		case 'i':
-			action = ACTION_ENABLE_NVMET_PORT;
-			sprintf(host, "%s", optarg);
-			break;
-		case 'j':
-			json_flag = TRUE;
-			break;
-		case 'L':
-			action = ACTION_LOCK;
-			sprintf(device, "%s", optarg);
+			case 'h':
+				printf("%s", header);
+				online_menu(PROCESS);
```

Comment:
I still prefer the use of argvin[0] for one reason:
If someone changes the binary name to (for example) `rapiddisk-util`, the online_menu will show examples with the executed name.

Reply:
The problem rises when the specified command argument is something like this, and the messages get messed up:

```bash
shub@localhost:/tmp/rapiddisk-8.2.0/src$ sudo ../../test/.././test/subdir/.././subdir/bin/./testbin/rapiddisk -h
../../test/.././test/subdir/.././subdir/bin/./testbin/rapiddisk 8.2.0
Copyright 2011 - 2022 Petros Koutoupis

../../test/.././test/subdir/.././subdir/bin/./testbin/rapiddisk is an administration tool to manage RapidDisk RAM disk devices and
        RapidDisk-Cache mappings.

Usage: ../../test/.././test/subdir/.././subdir/bin/./testbin/rapiddisk [ -h | -v ] function [ parameters ]

Description:
        ../../test/.././test/subdir/.././subdir/bin/./testbin/rapiddisk is a RapidDisk module management tool to manage RapidDisk
        RAM disk devices. Dynamically create, remove, resize RAM volumes and if
        desired, map or unmap them as a cache volume to any block device.
[...[
```

If you want I rollback the change!

Reply:
No. Keep the change then. You make a great point.